### PR TITLE
Type improvement behavior correction, index_sample API already support float16, logits no need to cast. 

### PR DESCRIPTION
**Pull request by zxcd**
### PR types

Bug fixes
### PR changes

Others
### Description

Type improvement behavior correction, index_sample API already support float16, logits no need to cast.

### CODE REVIEW

**Review by wj-Mcat — APPROVED** [review]
LGTM
